pyriemann/utils/mean.py

```diff
@@ -733,3 +734,144 @@ def nanmean_riemann(covmats, tol=10e-9, maxiter=100, init=None,
         sample_weight=sample_weight
     )
     return C
+
+
+###############################################################################
+
+
+def median_euclid(covmats, tol=10e-6, maxiter=50, init=None):
+    """Median of matrices according to the Euclidean metric.
+
+    Return the Euclidean geometric median for matrices,
+    different from the marginal median provided by numpy.median().
+
+    Parameters
+    ----------
+    covmats : ndarray, shape (n_matrices, n_channels, n_channels)
+        Set of matrices.
+    tol : float, default=10e-6
+        The tolerance to stop the gradient descent.
+    maxiter : int, default=50
+        The maximum number of iterations.
+    init : None | ndarray, shape (n_channels, n_channels), default=None
+        A matrix used to initialize the gradient descent.
+        If None, the Euclidean mean is used.
+
+    Returns
+    -------
+    C : ndarray, shape (n_channels, n_channels)
+        Euclidean geometric median.
+
+    Notes
+    -----
+    .. versionadded:: 0.3.1
+
+    References
+    ----------
+    .. [1] Vardi Y and Zhan C-H. "The multivariate L1-median and associated
+        data depth", PNAS, 2000
+    """
+    if init is None:
+        C = mean_euclid(covmats)
+    else:
+        C = init
+
+    for _ in range(maxiter):
+        dists = pairwise_distance(
+            covmats,
+            C[np.newaxis, ...],
+            metric='euclid'
+        )[:, 0]
+        is_zero = (dists == 0)
+
+        inv_dists = 1 / dists[~is_zero]
+        weights = inv_dists / np.sum(inv_dists)
+        Cnew = np.einsum('a,abc->bc', weights, covmats[~is_zero])
```

Comment:
how about rather adding weights param to mean_euclid and iterate calls to mean_euclid?

Reply:
Ok. I must read papers again.